### PR TITLE
return generated keys optionally

### PR DESCRIPTION
**Pull request by oridag**

### CODE REVIEW

**Comment by oridag** [comment]
Adding some context here: Snowflake JDBC driver asserts that `autoGeneratedKeys` is set to `Statement.NO_GENERATED_KEYS`, so this change is required for Snowflake use

**Comment by oridag** [comment]
@jdegoes merged from master and fixed lint errors

File: core/src/main/scala/zio/jdbc/SqlFragment.scala [hunk 1]
```diff
@@ -190,8 +190,8 @@ sealed trait SqlFragment { self =>
    * parsed and returned as `Chunk[Long]`. If keys are non-numeric, a
    * `Chunk.empty` is returned.
    */
-  def insert: ZIO[ZConnection, Throwable, UpdateResult] =
-    ZIO.scoped(executeWithUpdateResult(self))
+  def insert(returnAutoGeneratedKeys: Boolean = true): ZIO[ZConnection, Throwable, UpdateResult] =
```
[the remark this answers is not in the document]
Comment (oridag) [reply core/src/main/scala/zio/jdbc/SqlFragment.scala]: done

**Comment by oridag** [comment]
@jdegoes can you please re-review?

**Comment by oridag** [comment]
@jdegoes / @adamgfraser any chance we can move forward with merging this?

**Comment by nightscape** [comment]
I also just ran into this. @jdegoes / @adamgfraser would you have time for another review?

**Review by jdegoes — APPROVED** [review]

**Pull request merged**